bundlewrap/utils/__init__.py

```diff
@@ -96,21 +105,20 @@ def __add__(self, other):
         if isinstance(other, Fault):
             def callback():
                 return self.value + other.value
-            return Fault(callback)
+            return Fault(self.id_list + other.id_list, callback)
         else:
             def callback():
                 return self.value + other
-            return Fault(callback)
+            return Fault(self.id_list + ['raw {}'.format(other)], callback)
```

Comment:
```suggestion
            return Fault(self.id_list + ['raw {}'.format(repr(other))], callback)
```

Otherwise you cannot tell the difference between `other` being  `1` or `"1"`.

Reply:
not a fan of the `repr()` here, but it seems the simplest way to catch most issues

Reply:
Applied.